lib/templates.js

```diff
@@ -439,10 +439,15 @@ ContextClosure.prototype.attachTo = function(parent, node, context) {
   return this.template.attachTo(parent, node, closureContext);
 };
 ContextClosure.prototype.dependencies = function(context, options) {
-  if (DependencyOptions.shouldIgnoreTemplate(this, options)) return;
+  if (DependencyOptions.shouldIgnoreTemplate(this.template, options)) return;
   var closureContext = context.closureChild(this.context);
   return this.template.dependencies(closureContext, options);
 };
+ContextClosure.prototype.equals = function(other) {
+  return (other instanceof ContextClosure) &&
+    (other.template === this.template) &&
```

Comment:
Do we need to worry about `this.template` or `other.template` themselves being ContextClosures?

Reply:
Good idea. Let's recursively use the equals function